components/hierarchical-view/hierarchical-view-mixin.js

```diff
@@ -261,16 +279,20 @@ export const HierarchicalViewMixin = superclass => class extends superclass {
 	}
 
 	__autoSize(view) {
-		if (this.childView || view.offsetParent === null) return;
+		if (this.__isAutoSized || this.childView) return;
```

Comment:
Logic mostly stays the same here, I just moved the short-circuit for `this.childView` up higher, added a new short-circuit so this only runs on the first intersection, and wrapped everything in a `rAF`.